### ISSUE TITLE
Duplicate function does not copy signals

### ISSUE THREAD

**Issue by Chaosus** [godotengine/godot#12951]
**Issue description:**
In https://github.com/godotengine/godot/pull/12893 I mentioned the function 'duplicate' is not working completly correctly( does not copy signals) and also suggest change, but later decided to leave this problem... for the issue topic, Because the change of this function may lead to various and unobvious bugs, and needed to review from most experienced users, before any merge..

(in node.cpp/duplicate_signals)
```
for (List<Connection>::Element *E = conns.front(); E; E = E->next()) {

		if (E->get().flags & CONNECT_PERSIST) {
			//user connected
			NodePath p = p_original->get_path_to(this);
			Node *copy = p_copy->get_node(p);

			Node *target = Object::cast_to<Node>(E->get().target);
			if (!target) {
				continue;
			}
			NodePath ptarget = p_original->get_path_to(target);
			Node *copytarget = p_copy->get_node(ptarget);

			if (copy && copytarget) {
				copy->connect(E->get().signal, copytarget, E->get().method, E->get().binds, CONNECT_PERSIST);
			}
		}
	}
```

the suggested change is replace that loop to -

```
 	for (List<Connection>::Element *E = conns.front(); E; E = E->next()) {
 		if (E->get().flags & CONNECT_PERSIST) {
 			node->connect(E->get().signal, E->get().target, E->get().method, E->get().binds, E->get().flags);
 		}
 	}

```
Discuss..

**Comment by RandomShaper** [comment]
Today I've been able to late review your recent work on signals. Since this thread is still open I'll replay here.

Regarding, #12893 the refactor is OK.

The code added in that PR to the editor to duplicate signals is the same you are proposing here, which opens some questions:
1. I think the upper piece of code aims to rewrite the paths to the target nodes relative to the copies of the nodes. Both in #12893 and in the lower code snippet that logic is missing.
2. Can you provide an example in which the lower code fragment correctly copies signals, while the upper one doesn't?

**Comment by Chaosus** [comment]
`Can you provide an example in which the lower code fragment correctly copies signals, while the upper one doesn't?`
You can do it yourself by creating node, connect its signal to something markable(for example change color of ColorRect when mouse entered), add script and then called duplicate on it. If you used my code fragment in duplicate_signals clones will duplicate signals, else - not. I dont want touch it myself because @reduz maked a change - https://github.com/godotengine/godot/commit/9b692b174b06ebb80d6e5a82309b9c9fc44c797d and I dont want to break his work, before any merge I decided to discuss drawback

**Comment by Chaosus** [comment]
Also look at https://github.com/godotengine/godot/issues/11182, this is related to this, and likely must be solved with this issue

**Comment by RandomShaper** [comment]
As long as the fix for #11182 is a refactor (and that seems to be the case) we can address this before or after that.

Could you kindly provide a ready-to-use test project? I'd like to help, but my time is very limited.

**Comment by Chaosus** [comment]
Okay, in this project if you press space it duplicate, but does not duplicate signal(mouse_entered) - 
[TestSignals.zip](https://github.com/godotengine/godot/files/1484437/TestSignals.zip)
